src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/ExcelAdaptorServiceImpl.java

```diff
@@ -114,23 +115,43 @@ private void validateHeaders(List<String> headersToBeValidated,
         Field rowField = getRowIdField((Class<Object>) classType);
         Optional<Object> bean;
         while (rowIterator.hasNext()) {
-            bean = handleRowProcessing(classType, rowField, headers, rowIterator.next(), parentFieldMap,
-                childHeaderToCellMap, customObjectFieldsMapping);
-            bean.ifPresent(o -> objectList.add((T) o));
+            Row row = rowIterator.next();
+            if (isNotTrue(isBlankRow(row))) {
+                bean = handleRowProcessing(classType, rowField, headers, row, parentFieldMap,
+                        childHeaderToCellMap, customObjectFieldsMapping);
+                bean.ifPresent(o -> objectList.add((T) o));
+            } else {
+                break;
```

Comment:
Is there any test to cover this scenario where we have empty rows in the middle of the records?

Reply:
yes, `parseXlsxWhichHasFormulaWithEmptyRows` in `ExcelAdaptorServiceImplTest` covers this scenario as I have modified the `src/test/resources/Staff Data Upload.xlsx` file and included a record leaving some blank rows in between and the test is passing. I did some manual testing too.

Reply:
ah ok :) Thanks for pointing it out